src/App.tsx

```diff
@@ -8,55 +8,163 @@ import { PostsList } from './components/PostsList';
+  };
+
+  useEffect(() => {
+    getUsers();
+  }, []);
+
+  const getUserPostsFromServer = async (userId: number) => {
+    setIsLoaderVisible(true);
+    setIsPostListVisible(false);
+
+    try {
+      const fetchPost = await customFetch.getPostFromServer(userId);
+
+      setErrorType(null);
+
+      setPosts(fetchPost);
+      setIsPostListVisible(true);
+      setErrorType(null);
+    } catch (error) {
+      setErrorType(ErrorType.POSTS);
+      setIsPostListVisible(false);
+    } finally {
+      setIsLoaderVisible(false);
+    }
+  };
+
+  const getCommentByPostFromServer = async (postId: number) => {
+    setIsCommentFetch(true);
+    setIsCommentListHidden(false);
+
+    try {
+      const fetchComments = await customFetch.getPostComments(postId);
+
+      setComments(fetchComments);
+      setIsPostListVisible(true);
+      setErrorType(null);
+    } catch (error) {
+      setErrorType(ErrorType.COMMENTS);
+    } finally {
+      setIsCommentFetch(false);
+    }
+  };
+
+  const handleOpenerDetails = (post: Post) => {
+    setSelectedPost(post);
+    getCommentByPostFromServer(post.id);
+    setOpenFormForComment(false);
+  };
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  select={selectedUser}
+                  onSelect={setSelectedUser}
+                  getUserPosts={getUserPostsFromServer}
+                  setIsCommentListHidden={setIsCommentListHidden}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoaderVisible ? (
+                  <Loader />
+                ) : (
+                  <>
+                    {(isErrorType && loadingErrors.includes(isErrorType)) && (
+                      <div
+                        className="notification is-danger"
+                        data-cy="PostsLoadingError"
+                      >
+                        {isErrorType}
+                      </div>
+                    )}
+                  </>
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {isPostListVisible && (
+                  <PostsList
+                    posts={posts}
+                    selectPost={selectedPost}
```

Comment:
The property name selectPost is a bit misleading, as it suggests an action. Consider renaming it to something more descriptive, like selectedPost.